tests/core/editor/destroy.js

```diff
@@ -1,84 +1,112 @@
-		bender.editorBot.create( {}, function( bot ) {
-			this.wait( function() {
-				var editor = bot.editor;
-				editor.destroy();
-				assert.areSame( '', editor.getSnapshot() );
+			element = CKEDITOR.document.getById( name );
+			this.editor.destroy();
+
+			editor = CKEDITOR.replace( element );
+			editor.destroy();
+
+			// initConfig is called asynchronously.
+			wait( function() {
+				warnStub.restore();
+				assert.isTrue( warnStub.calledOnce, 'CKEDITOR.warn should be called once.' );
+				assert.areEqual( 'editor-incorrect-destroy', warnStub.firstCall.args[ 0 ],
+					'CKEDITOR.warn error code should be "editor-incorrect-destroy".' );
 			}, 0 );
-		} );
-	},
-
-	'test destroy editor before it is fully initialized': function() {
-		var name = 'test_editor',
-			element,
-			editor,
-			warnStub = sinon.stub( CKEDITOR, 'warn' );
-
-		element = CKEDITOR.document.getById( name );
-		this.editor.destroy();
-
-		editor = CKEDITOR.replace( element );
-		editor.destroy();
-
-		// initConfig is called asynchronously.
-		wait( function() {
-			warnStub.restore();
-			assert.isTrue( warnStub.calledOnce, 'CKEDITOR.warn should be called once.' );
-			assert.areEqual( 'editor-incorrect-destroy', warnStub.firstCall.args[ 0 ],
-				'CKEDITOR.warn error code should be "editor-incorrect-destroy".' );
-		}, 0 );
-
-	},
-
-	'test check editable existence on blur': function() {
-		CKEDITOR.replace( 'focused', {
-			on: {
-				instanceReady: function( evt ) {
-					resume( function() {
-						var editor = evt.sender;
-						// Simulate the circumstances instead of creating them.
-						editor.focusManager.hasFocus = true;
-						sinon.stub( editor, 'editable' ).returns( null );
-						editor.focusManager.blur( 1 );
-						assert.pass();
-					} );
+
+		},
+
+		'test check editable existence on blur': function() {
+			CKEDITOR.replace( 'focused', {
+				on: {
+					instanceReady: function( evt ) {
+						resume( function() {
+							var editor = evt.sender;
+							// Simulate the circumstances instead of creating them.
+							editor.focusManager.hasFocus = true;
+							sinon.stub( editor, 'editable' ).returns( null );
+							editor.focusManager.blur( 1 );
+							assert.pass();
+						} );
+					}
 				}
-			}
-		} );
+			} );
+
+			wait();
+		},
+
+		// (#1722)
+		'test destroy attached filters': function() {
+			var filters = countFilters();
+			bender.editorBot.create( { name: 'editor_filter_destroy' }, function( bot ) {
+				var editor = bot.editor;
 
-		wait();
+				new CKEDITOR.filter( editor, 'br' );
+				new CKEDITOR.filter( editor );
+				new CKEDITOR.filter( 'br' );
+
+				editor.destroy();
+
+				assert.areEqual( 0, countFilters( editor ) );
+				assert.areEqual( filters + 1, countFilters() );
+			} );
+		}
+	} );
+
+	function countFilters( editor ) {
+		var filters = bender.tools.objToArray( CKEDITOR.filter.instances );
+		return editor ? filters.filter( function( filter ) {
```

Comment:
This function won't work in IE8, as there is no `Array.prototype.filter`. Our own `CKEDITOR.tools.array.filter` should be used here.